### PR TITLE
Add support for customizable action colors

### PR DESCRIPTION
**Pull request by micahmo**
## Pull Request Description

The PR adds the ability to customize the colors for the following actions in Thuder.
* Upvote
* Downvote
* Save
* Mark as read/unread
* Reply/Edit

These colors apply to swipe gestures, inkwells, status indicators, etc. Note that we can pretty easily add more colors in the future. It should also be possible for us to use a similar mechanism to customize other colors in the app (e.g., nested comment indicators or media badge indicators).

Finally, there is a new settings hint in the gestures page which points to this new setting.

As this was quite a big change, please be on the lookout for unusual colors!

## Issue Being Fixed

Issue Number: #1301 and #25

## Screenshots / Recordings

https://github.com/thunder-app/thunder/assets/7417301/59093405-8eec-4647-a481-7c72e18d5010

## Checklist

- [ ] Did you update CHANGELOG.md?
- [ ] Did you use localized strings where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

File: lib/settings/pages/theme_settings_page.dart [hunk 1]
```diff
@@ -364,6 +400,250 @@ class _ThemeSettingsPageState extends State<ThemeSettingsPage> {
                       ],
                     ),
                   ),
+                  Container(
```
Comment (hjiangsu) [inline lib/settings/pages/theme_settings_page.dart:403]: It might be good to move this to a separate widget since it's quite large

**Review by hjiangsu — APPROVED** [review]
LGTM, just one comment!

Could you also add in red as a colour option? This request comes from https://lemmy.world/comment/9492460

**Comment by micahmo** [comment]
> Could you also add in red as a colour option? This request comes from https://lemmy.world/comment/9492460

Done!

https://github.com/thunder-app/thunder/assets/7417301/d83c5ff1-c304-47a2-a27c-863f19510a3e

File: lib/settings/pages/theme_settings_page.dart (hunk 1 above)
Comment (micahmo) [reply lib/settings/pages/theme_settings_page.dart]: Done! This should be good to go now.

**Pull request merged**